lib/command.js

```diff
@@ -40,6 +42,16 @@ export const parseCommand = command => {
 	return tokens;
 };
 
+const parseExpression = expression => {
+	if (typeof expression === 'object' && expression !== null && 'stdout' in expression) {
```

Comment:
```suggestion
	if (typeof expression === 'object' && expression !== null && expression.stdout !== undefined) {
```

I could be wrong, but I believe when `options.stdout` is `inherit`, `expression` would have a `stdout` key, but its value would be `undefined` (as opposed to having no `stdout` key).

Reply:
Here's my WIP regarding this and discussions [r1125573007](https://github.com/sindresorhus/execa/pull/528#discussion_r1125573007), [r1125572721](https://github.com/sindresorhus/execa/pull/528#discussion_r1125572721):
```js
const parseExpression = expression => {
	if (typeof expression === 'object' && expression !== null) {
		if ('stdout' in expression) {
			if (Buffer.isBuffer(expression.stdout)) {
				return expression.stdout.toString();
			}

			if (typeof expression.stdout === 'string') {
				return expression.stdout;
			}

			throw new TypeError(`Unexpected '${typeof expression.stdout}' stdout in template expression`);
		}

		if (expression instanceof Promise) {
			throw new TypeError('Unexpected promise in template expression');
		}

		throw new TypeError('Unexpected object in template expression');
	}

	return String(expression);
};
```

Although, I'm unsure if this is too exhaustive as we have been coercing all values thus far:
https://github.com/sindresorhus/execa/blob/eae327cc0a49f61b17f699b9199c7953e1497c1a/test/command.js#L144-L147

Or perhaps, we should lean into this, remove the coercion, and additionally validate the standalone `string` and `number` expressions (which would be more representative of the types). Thoughts?

Reply:
Yes, I was also thinking of the fact that we previously just relied on the coercion instead of failing hard. :thinking: 

Personally, I tend to favor failing hard so that users realize their errors early. In this case, if a user forgets to call `await`, or if a user tries to use this feature with a process with `options.stdout: 'inherit'`, this is a mistake, so it would be helpful to report it so users can fix it. What do you think?

Reply:
Agreed 👍  Here's my first pass at removing invalid template expression coercion and introducing error handling: https://github.com/sindresorhus/execa/pull/528/commits/03e7ab9a79c88e50c306a928a9e59e229d87c2df